src/test/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParserTest.java

```diff
@@ -137,17 +122,40 @@ public void parseAndStoreTranscript_persistDataInDatastoreForShortVideo() throws
 
   @Test
   public void parseAndStoreTranscript_persistDataInDatastoreForLongVideo() throws IOException {
-    when(request.getParameter(LectureUtil.VIDEO_ID)).thenReturn(LONG_VIDEO_ID);
-    when(request.getParameter(LectureUtil.ID)).thenReturn(LECTURE_ID_C);
     Key lectureKeyC = KeyFactory.createKey(LectureUtil.KIND, Long.parseLong(LECTURE_ID_C));
 
-    TranscriptParser.getParser().parseAndStoreTranscript(LONG_VIDEO_ID, lectureKeyC);
+    TranscriptParser.getParser().parseAndStoreTranscript(
+        LONG_VIDEO_ID, lectureKeyC, /* transcriptLanguage= */ "en");
 
     int actualQueryCount = entitiesInDatastoreCount(lectureKeyC);
     int expectedQueryCount = (longVideoTranscriptLines).size();
     assertThat(actualQueryCount).isEqualTo(expectedQueryCount);
   }
 
+  @Test
+  public void parseAndStoreTranscript_invalidLanguage_throwsException() throws IOException {
+    Key lectureKeyB = KeyFactory.createKey(LectureUtil.KIND, Long.parseLong(LECTURE_ID_B));
+
+    try {
+      TranscriptParser.getParser().parseAndStoreTranscript(
+        LONG_VIDEO_ID, lectureKeyB, /* transcriptLanguage= */ "notAValidLanguage");
+      fail();
+    } catch (IOException e) {
+    }
+  }
+
+  @Test
+  public void parseAndStoreTranscript_noLanguageAvailable_throwsException() throws IOException {
```

Comment:
can you make these both `throws Exception` b/c they are tests

Reply:
thanks for catching that, updated!